distributed/shuffle/tests/test_graph.py

```diff
@@ -65,18 +71,25 @@ def test_raise_on_sparse_data():
     df = dd.from_pandas(
         pd.DataFrame({"x": pd.array(range(10), dtype="Sparse[float64]")}), npartitions=5
     )
-    with pytest.raises(
-        TypeError, match="p2p does not support sparse data"
-    ), dask.config.set({"dataframe.shuffle.method": "p2p"}):
-        df.shuffle("x")
+    if dd._dask_expr_enabled():
+        df.shuffle("x").optimize()
```

Comment:
From what I understand, this should raise.

Reply:
Yep, and this is the other. Maybe xfailing these is better until updated in dask-expr. Looking at them again now like this seems not great.

Reply:
https://github.com/dask-contrib/dask-expr/pull/951